src/main/java/org/generationcp/middleware/dao/GermplasmSearchDAO.java

```diff
@@ -1358,16 +1383,24 @@ private boolean addPreFilteredGids(final GermplasmSearchRequest germplasmSearchR
 			while (iterator.hasNext()) {
 				final Map.Entry<String, String> entry = iterator.next();
 				queryBuilder.append(String.format("a.gid in (select a.gid from atributs a \n"
-					+ " inner join udflds u on a.atype = u.fldno \n"
-					+ " where u.fcode = :attributeKey%s and aval like :attributeValue%<s )", entry.getKey()));
+					+ " INNER JOIN cvterm cv on a.atype = cv.cvterm_id \n"
+					+ " INNER JOIN cvtermprop cp ON cp.type_id = " + TermId.VARIABLE_TYPE.getId() + " and cv.cvterm_id = cp.cvterm_id "
```

Comment:
Same here, move the filtering of variable types to main clause instead of where clause:
`INNER JOIN cvterm vartype on vartype.cvterm.name = cp.value and vartype.cvterm_id in (...)`

Reply:
done.